dashboard/src/components/Table/TreeTable.tsx

```diff
@@ -57,57 +57,78 @@ const treeTableHeaders = headerLabelOrElement.map(item =>
   typeof item === 'string' ? <FormattedMessage key={item} id={item} /> : item,
 );
 
+/*const LinkItem: React.FC<PropsWithChildren<{ target: string, row: TreeTableBody, unsafeOrigin: string | undefined }>> =
+({ children, target, row, unsafeOrigin }) => {
+  const [linkTarget, setLinkTarget] = useState<string | undefined>(undefined);
+  const linkRef = useRef<HTMLAnchorElement>(null);
+
+  const origin = zOrigin.parse(unsafeOrigin);
+
+  
+
+  useEffect(() => {
+    if(linkTarget) {
+      setLinkTarget(undefined);
+      linkRef.current?.click();
+    }
+  }, [linkTarget]);
+
+  return (
+    <Link
+      {...linkProps}
+      ref={linkRef}
+      target={linkTarget}
+      onContextMenu={(e) => {
+        e.preventDefault();
+        setLinkTarget("_blank");
+      }}
+    >
+      {children}
+    </Link>
+  );
+}*/
+
 const TreeTableRow = (row: TreeTableBody): JSX.Element => {
   const { origin: unsafeOrigin } = useSearch({ strict: false });
   const origin = zOrigin.parse(unsafeOrigin);
 
-  const navigate = useNavigate({ from: '/tree' });
-
-  const navigateToTreeDetailPage = useCallback(
-    (event: React.MouseEvent<HTMLTableCellElement>) => {
-      const el = event.currentTarget;
-      const target = el.getAttribute('data-target');
-
-      const safeTarget = zPossibleValidator.parse(target);
-
-      navigate({
-        to: '/tree/$treeId',
-        params: { treeId: row.id },
-        search: {
-          tableFilter: {
-            bootsTable: possibleTestsTableFilter[0],
-            buildsTable: possibleBuildsTableFilter[2],
-            testsTable: possibleTestsTableFilter[0],
-          },
-          origin: origin,
-          currentTreeDetailsTab: safeTarget,
-          diffFilter: {},
-          treeInfo: {
-            gitUrl: row.url,
-            gitBranch: row.branch,
-            treeName: row.tree_name ?? undefined,
-            commitName: row.commitName,
-            headCommitHash: row.id,
-          },
-        },
-      });
+  // const navigate = useNavigate({ from: '/tree' });
+
+  const linkProps = (
+    target: string,
+  ): LinkProps & { style: React.CSSProperties } => ({
+    to: '/tree/$treeId',
+    params: { treeId: row.id },
+    style: {
+      width: '100%',
+      display: 'inline-block',
+      height: '100%',
     },
-    [
-      navigate,
-      row.id,
-      row.url,
-      row.branch,
-      row.tree_name,
-      row.commitName,
-      origin,
-    ],
-  );
+    search: {
+      tableFilter: {
+        bootsTable: possibleTestsTableFilter[0],
+        buildsTable: possibleBuildsTableFilter[2],
+        testsTable: possibleTestsTableFilter[0],
+      },
+      origin: origin,
+      currentTreeDetailsTab: zPossibleValidator.parse(target),
+      diffFilter: {},
+      treeInfo: {
+        gitUrl: row.url,
+        gitBranch: row.branch,
+        treeName: row.tree_name ?? undefined,
```

Comment:
i think that you don't need the `??` here, considering it will already be undefined or null if there is no tree_name

Reply:
ditto